### PR TITLE
Pod logs

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HAC-3600

## Description

- Adds Pod Logs component and related utils
- Updates Component list view and adds `view pod logs` action
- Updates `view logs` to `view build logs` in ComponentListView action and BuildLogs modal
- Updates fetching of GitOpsDeployment using SnapshotEncBinding as existing in-place fetch of GitOpsDeployment fails due to missing labels
- Adds related unit tests

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2023-04-10 at 8 23 49 PM

Screenshot 2023-04-10 at 8 32 51 PM
Screenshot 2023-04-10 at 8 17 37 PM
<img width="1494" alt="Screenshot 2023-04-04 at 6 45 17 PM" src="https://user-images.githubusercontent.com/24852534/229879733-1dd2b06d-0168-4f42-8ded-7f03547ab15a.png"

## How to test or reproduce?

Added unit tests

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
component list page crashes, when a new component is added.

**Comment by karthikjeeyar** [comment]
Deployment is successful and the pod is running bu thte Podstatus says `PodScheduled`, shouldn't it be in `Running` state?

image

File: src/components/Components/ComponentListItem.tsx [hunk 1]
```diff
@@ -63,6 +70,24 @@ export const ComponentListItem: React.FC<ComponentListViewItemProps> = ({
...
+    namespace: deploymentResource?.namespace,
+  });
+
+  const podSelector = React.useMemo(
+    () => !deploymentLoadError && deploymentLoaded && deployment.spec?.selector,
+    [deploymentLoaded, deployment, deploymentLoadError],
+  );
+
+  // const [componnentState, setComponentState] = React.useState<{ [name: string]: PACState }>({});
+
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply src/components/Components/ComponentListItem.tsx]: removed

File: src/components/Components/BuildStatusColumn.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const BuildStatusColumn: React.FC<BuildStatusComponentProps> = ({ component }) =
             data-testid={`view-build-logs-${component.metadata.name}`}
             isInline
           >
-            View logs
+            View build logs
```
[the remark this answers is not in the document]
Comment (abhinandan13jan) [reply src/components/Components/BuildStatusColumn.tsx:36]: updated
Screenshot 2023-04-10 at 8 32 40 PM

**Comment by abhinandan13jan** [comment]
@karthikjeeyar updated contional checking for GitOpsDeployment so now add component should work fine and also updated modal to say `view build logs`

**Review by karthikjeeyar — COMMENTED** [review]
When you expand the logs to full screen, the nested navigation menu is appearing in the UI.


Screenshot 2023-04-11 at 9 33 38 PM

**Review by rottencandy — APPROVED** [review]
/lgtm

**Pull request merged**